src/adapter/src/catalog/builtin_table_updates.rs

```diff
@@ -1168,15 +1169,55 @@ impl CatalogState {
                 ServiceProcessMetrics {
                     cpu_nano_cores,
                     memory_bytes,
+                    disk_usage_bytes,
+                    disk_capacity_bytes,
                 },
             )| {
+                // We only populate a `disk_usage` if we have all 5 of of:
+                // - a disk limit (so it must be an actual managed cluster with a real limit)
+                // - a reported disk capacity
+                // - a reported disk usage
+                // - disk capacity is less than disk_usage.
+                // - There are no overflows
+                //
+                //
+                // The `disk_usage` is augmented with the difference between
+                // the limit and the `capacity`, so users don't erroneously think they
+                // have more real usable capacity than they actually do.
+                let disk_usage = if let (Some(disk_usage), Some(disk_capacity), Some(disk_limit)) =
+                    (disk_usage_bytes, disk_capacity_bytes, disk_limit)
+                {
+                    if disk_limit.0 .0 >= *disk_capacity {
+                        if let Some(disk_usage) =
+                            (disk_limit.0 .0 - *disk_capacity).checked_add(*disk_usage)
+                        {
+                            Datum::UInt64(disk_usage)
+                        } else {
+                            tracing::warn!(
+                                "disk usage  {} + correct {} ?",
+                                disk_usage,
+                                disk_limit.0 .0 - *disk_capacity
+                            );
```

Comment:
these should be errors IMO

Reply:
I just meant make this (and the similar one below) `tracing::error` instead of `tracing::warn` -- is that what you're saying would be risky? If so, why?

Reply:
`error`'s trigger sentry events, not sure we want those for this right now?

Reply:
Yeah that was my point. I think it's desirable to raise sentry events if we're losing data in this table as it would mean our assumptions about what the values mean is wrong and therefore the whole contents of the table could be wrong. Like it shouldn't page someone but I think it should be looked at with some urgency. But I accepted the PR regardless, so it's up to you!

Reply:
you make a good point, ill set it to error!